drivers/ps2/ps2_mchp_xec.c

```diff
@@ -205,14 +253,82 @@ static int ps2_xec_enable_interface(const struct device *dev)
 	struct ps2_xec_data * const data = dev->data;
 	struct ps2_regs * const regs = config->regs;
 
-	ps2_xec_girq_clr(dev);
+	ps2_xec_girq_clr(config->girq_id, config->girq_bit);
 	regs->CTRL = MCHP_PS2_CTRL_EN;
 
 	k_sem_give(&data->tx_lock);
 
 	return 0;
 }
 
+#ifdef CONFIG_PM_DEVICE
+static int ps2_xec_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	const struct ps2_xec_config *const devcfg = dev->config;
+	struct ps2_regs * const regs = devcfg->regs;
+	int ret = 0;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		if (devcfg->wakeup_source) {
+			/* Disable PS2 wake interrupt
+			 * Disable interrupt on PS2DAT pin
+			 */
+			if (devcfg->wakerx_gpio.port != NULL) {
+				ret = gpio_pin_interrupt_configure_dt(
+						&devcfg->wakerx_gpio,
+						GPIO_INT_DISABLE);
+				if (ret < 0) {
+					LOG_ERR("Fail to disable PS2 wake interrupt (ret %d)", ret);
+					return ret;
+				}
+			}
+			ps2_xec_girq_dis(devcfg->girq_id_wk, devcfg->girq_bit_wk);
+			ps2_xec_girq_clr(devcfg->girq_id_wk, devcfg->girq_bit_wk);
+		} else {
+#ifdef CONFIG_PINCTRL
+			ret = pinctrl_apply_state(devcfg->pcfg, PINCTRL_STATE_DEFAULT);
+#endif
+			regs->CTRL |= MCHP_PS2_CTRL_EN;
+		}
+	break;
+	case PM_DEVICE_ACTION_SUSPEND:
+		if (devcfg->wakeup_source) {
+			/* Enable PS2 wake interrupt
+			 * Configure Falling Edge Trigger interrupt on PS2DAT pin
+			 */
+			ps2_xec_girq_clr(devcfg->girq_id_wk, devcfg->girq_bit_wk);
+			ps2_xec_girq_en(devcfg->girq_id_wk, devcfg->girq_bit_wk);
+			if (devcfg->wakerx_gpio.port != NULL) {
+				ret = gpio_pin_interrupt_configure_dt(
+					&devcfg->wakerx_gpio,
+					GPIO_INT_MODE_EDGE | GPIO_INT_TRIG_LOW);
+				if (ret < 0) {
+					LOG_ERR("Fail to enable PS2 wake interrupt(ret %d)", ret);
+					return ret;
+				}
+			}
+		} else {
+			regs->CTRL &= ~MCHP_PS2_CTRL_EN;
+#ifdef CONFIG_PINCTRL
```

Comment:
Now that pinmux is gone do we still these kind of checks?

Reply:
CONFIG_PINCTRL is still a configurable option in dts hence keeping it.

Reply:
This is no longer the case, please drop as the platform only works with pinctrl now. Driver must select pinctrl.

Reply:
Ok, CONFIG_PINCTRL has been removed.